src/Websockettest.js

```diff
@@ -554,53 +573,73 @@ const RMBTTest = (function() {
 
     /**
      * Download n Chunks from the test server
+     * @param {RMBTTestThread} thread containing an open socket
      * @param {Number} total how many chunks to download
-     * @param {RMBTThread} thread containing an open socket
-     * @param {Callback} onsuccess expects one argument (String)
+     * @param {Number} chunkSize size of single chunk in bytes
+     * @param {Function} onsuccess expects one argument (String)
      */
     function downloadChunks(thread, total, chunkSize, onsuccess) {
         //console.log(String.format(Locale.US, "thread %d: getting %d chunk(s)", threadId, chunks));
         let socket = thread.socket;
         let remainingChunks = total;
         let expectBytes = _chunkSize * total;
         let totalRead = 0;
+        let lastBuffer = null;
+
+        // https://stackoverflow.com/questions/33702838/how-to-append-bytes-multi-bytes-and-buffer-to-arraybuffer-in-javascript
```

Comment:
I did not notice any substantial changes in my local environment. I think I reverted the code once and checked against the dev server. The result was the same.
Though some minor improvements should still be possible.